doc/cabal-package-description-file.rst

```diff
@@ -2693,87 +2693,137 @@ Starting with Cabal-2.2 it's possible to use common build info stanzas.
 
-There are two kinds. You can specify one or the other or both at once:
+Here are some examples that point to snapshots of the Cabal library in
+its Git repository:
 
--  The ``head`` kind refers to the latest development branch of the
-   package. This may be used for example to track activity of a project
-   or as an indication to outside developers what sources to get for
-   making new contributions.
-
--  The ``this`` kind refers to the branch and tag of a repository that
-   contains the sources for this version or release of a package.  For most
-   source control systems this involves specifying a tag, id or hash of some
-   form and perhaps a branch.
+::
 
-As an example, here are the repositories for the Cabal library. Note that the
-``this`` kind of repository specifies a tag.
+    -- Snapshot via Git commit hash
+    source-repository this
+      type:     git
+      location: https://github.com/haskell/cabal
+      tag: 260ecdc3d848782d4df49e629cb0a5dc9e96ca9e
 
-::
+    -- Snapshot via Git tag
+    source-repository this
+      type:     git
+      location: https://github.com/haskell/cabal
+      tag:      cabal-install-v3.12.1.0
 
+    -- Snapshot via latest commit (head) of the default (master) branch
     source-repository head
       type:     git
       location: https://github.com/haskell/cabal
 
-    source-repository this
+    -- Snapshot via latest commit (head) of the specified branch
+    source-repository head
       type:     git
       location: https://github.com/haskell/cabal
-      tag:      1.6.1
+      branch:   3.12
+
+    -- Snapshot via latest commit (head) in subdirectory of a monorepo
+    source-repository head
+      type:     git
+      location: https://github.com/haskell/vector.git
+      subdir:   vector
+
+Note the two notation variants ``this`` and ``head``:
+
+-  The ``this`` variants refers to a specific snapshot of a repository
+   that contains the sources for this version or release of your package.
+   For most source control systems this involves specifying a tag, id or
+   hash of some commit and perhaps a branch.
+
+-  The ``head`` variant refers to the latest source code snapshot
+   in a repository branch of your package. This may be used for example
+   to track activity of a project or as an indication to outside
+   developers what sources to get for making new contributions.
 
 The :ref:`cabal get<cabal-get>` command uses the kind of repository with
 its ``--source-repository`` option, if provided.
 
 .. _source-repository-fields:
 
-The :ref:`VCS fields<vcs-fields>` of ``source-repository`` are:
-
-..
-  data SourceRepo = SourceRepo
-    { repoKind :: RepoKind
-    , repoType :: Maybe RepoType
-    , repoLocation :: Maybe String
-    , repoModule :: Maybe String
-    , repoBranch :: Maybe String
-    , repoTag :: Maybe String
-    , repoSubdir :: Maybe FilePath
-    }
+The fields of ``source-repository`` are:
 
-.. pkg-field:: type: VCS kind
+.. pkg-field:: type: VCS tool
```

Comment:
I don't think `tool` is a good word here. Kind is better but it's overloaded in this section referring to this or head. I'd use "type" for one of the cases and "kind" for another.

Reply:
How about "The VCS hosting the repository"?

Reply:
I feel like it has to be one word. I also don't think "hosting" is accurate: I'm used to think of hosting in the context of VCS as the github hosting, gitlab hosting and others. So, to me using the word hosting here would be confusing.